src/utils/deploy/hash-fns.js

```diff
@@ -7,9 +7,63 @@ const pump = promisify(require('pump'))
 
 const { hasherCtor, manifestCollectorCtor } = require('./hasher-segments')
 
+// Maximum age of functions manifest (2 minutes).
+const MANIFEST_FILE_TTL = 12e4
+
+const getFunctionZips = async ({ directories, functionsConfig, manifestPath, rootDir, statusCb, tmpDir }) => {
+  statusCb({
+    type: 'functions-manifest',
+    msg: 'Looking for a functions manifest file...',
+    phase: 'start',
+  })
+
+  if (manifestPath) {
+    try {
+      // eslint-disable-next-line import/no-dynamic-require, node/global-require
+      const { functions, timestamp } = require(manifestPath)
+      const manifestAge = Date.now() - timestamp
+
+      if (manifestAge > MANIFEST_FILE_TTL) {
+        throw new Error('Manifest expired')
+      }
+
+      statusCb({
+        type: 'functions-manifest',
+        msg: 'Using bundled functions from manifest file (use --bundle to override)',
+        phase: 'stop',
+      })
+
+      return functions
+    } catch (error) {
+      statusCb({
```

Comment:
Should this be split into two separate errors handlers: one if `require()` fails, and one if manifest expired?

Reply:
Since it's not a fatal error and not something actionable by the user (other than using `--bundle`), I personally don't see the need to add too much detail, especially as it would complicate the logic a little bit.

I'm happy to do it if you feel strongly about it, though.

Reply:
This makes sense :+1: